### PR TITLE
Added support for using ROS parameters to spawn entities in Gazebo using ros_gz_sim::create

### LINKED ISSUE #459: Make `ros_gz_sim::create` use ROS parameters
Currently, the `ros_gz_sim::create` node uses gflags to configure where and how models are loaded.

For maximum compatibility with ROS, I think it makes sense to switch the gflags arguments to use rclcpp::Parameters.

### PR DESCRIPTION
**Pull request by Ayush1285**
# 🎉 New feature

Closes #459 

## Summary
Added ROS parameters support in ros_gz_sim::create node, while preserving the existing behavior. ROS parameters will be checked first, if not found then will revert back to Gflags arguments. 

## Test it
Launch the demo launch file from the ros_gz_sim_demos package. 
`ros2 launch ros_gz_sim_demos robot_description_publisher.launch.py`

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by Ayush1285** [comment]
@ahcorde can you please review it for approval?

**Comment by Ayush1285** [comment]
@ahcorde approval required for the workflow, can you please look into it?

**Comment by azeey** [comment]
Thanks for the contribution, @Ayush1285! Is this a duplicate of #463?

**Comment by Ayush1285** [comment]
> Thanks for the contribution, @Ayush1285! Is this a duplicate of #463?

Just checked the other PR. I guess we both created PRs parallelly for the same issue.

**Comment by azeey** [comment]
@Ayush1285 , I've closed the other PR. Let's iterate on this one and get it merged.

File: ros_gz_sim/src/create.cpp [hunk 1]
```diff
@@ -44,12 +48,48 @@ DEFINE_double(R, 0, "Roll component of initial orientation, in radians.");
...
-// ROS interface for spawning entities into Gazebo.
-// Suggested for use with roslaunch and loading entities from ROS param.
-// If these are not needed, just use the `gz service` command line instead.
+// Utility Function to avoid code duplication
+
+bool set_XML_from_topic(
+  const std::string & topic_name, const rclcpp::Node::SharedPtr ros2_node,
+  gz::msgs::EntityFactory & req)
+{
+  const auto timeout = std::chrono::seconds(1);
```
Comment (ahcorde) [inline ros_gz_sim/src/create.cpp:57]: include `<chrono>`, `<future>`, `<functional>`

File: ros_gz_sim/src/create.cpp [hunk 2]
```diff
@@ -44,12 +48,48 @@ DEFINE_double(R, 0, "Roll component of initial orientation, in radians.");
...
+    return true;
+  } else {
+    RCLCPP_ERROR(
+      ros2_node->get_logger(), "Failed to get XML from topic [%s].", topic_name.c_str());
+    return false;
+  }
+}
+
+
 int main(int _argc, char ** _argv)
```
Comment (ahcorde) [inline ros_gz_sim/src/create.cpp:90]:
```suggestion
}

int main(int _argc, char ** _argv)
```

File: ros_gz_sim/src/create.cpp [hunk 3]
```diff
@@ -59,8 +99,24 @@ int main(int _argc, char ** _argv)
                        [-R ROLL] [-P PITCH] [-Y YAW])");
   gflags::ParseCommandLineFlags(&_argc, &_argv, true);
 
+  // Declare ROS Parameters to be passed from Launch file
```
Comment (ahcorde) [inline ros_gz_sim/src/create.cpp:102]:
```suggestion
  // Declare ROS 2 parameters to be passed from launch file
```

File: ros_gz_sim_demos/launch/robot_description_publisher.launch.py [hunk 4]
```diff
@@ -76,14 +76,21 @@ def generate_launch_description():
...
+        parameters=[
+            {
+                'topic': '/robot_description'
+            }
+        ],
+        arguments=[
+            '-name', 'my_custom_model',
+            '-x', '1.2',
+            '-y', '3.3',
+            '-z', '2.4'
```
Comment (ahcorde) [inline ros_gz_sim_demos/launch/robot_description_publisher.launch.py:91]: these arguments are parameters now, aren't they ?

**Review by ahcorde — CHANGES_REQUESTED** [review]
do you mind to target `rolling` ? then we can backport the changes to `humble` and `iron`.

**Review by ahcorde — CHANGES_REQUESTED** [review]
Remove gflags on rolling too

**Comment by azeey** [comment]
> Remove gflags on rolling too

Why are we removing gflags? Wouldn't this break a lot of launch files already out there?

**Comment by ahcorde** [comment]
maybe add a warning

File: ros_gz_sim_demos/launch/robot_description_publisher.launch.py (hunk 4 above)
Comment (Ayush1285) [reply ros_gz_sim_demos/launch/robot_description_publisher.launch.py:91]: > these arguments are parameters now, aren't they ?

Yes, I'll move them to parameters.

**Comment by Ayush1285** [comment]
> do you mind to target `rolling` ? then we can backport the changes to `humble` and `iron`.

Yes, I've created a new PR #475 for rolling.

**Comment by ahcorde** [comment]
I will close this PR in favour of [this PR on rolling](https://github.com/gazebosim/ros_gz/pull/475)

**Pull request closed without merging**